rust/automerge/src/op_set/op.rs

```diff
@@ -0,0 +1,460 @@
+use crate::clock::Clock;
+use crate::exid::ExId;
+use crate::op_set::OpSetData;
+use crate::text_value::TextValue;
+use crate::types::{self, ActorId, ElemId, Key, ListEncoding, ObjId, OpId, OpIds, OpType};
+use crate::value::{Counter, ScalarValue, Value};
+use std::borrow::Cow;
+use std::cmp::Ordering;
+use std::collections::HashSet;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub(crate) struct OpIdx(u32);
+
+impl OpIdx {
+    pub(crate) fn new(index: usize) -> Self {
+        Self(index as u32)
+    }
+
+    pub(crate) fn get(&self) -> usize {
+        self.0 as usize
+    }
+
+    pub(crate) fn as_op2(self, osd: &OpSetData) -> Op2<'_> {
```

Comment:
Rather than having this `as_op2` method, can we instead have a method on `OpSetData::as_op_ref(idx: OpIdx)`. I feel like that would better communicate that the result is a reference to something indexed by `OpIdx` (see my other comment about renaming `Op2 -> OpRef`).